processor/processor.go

```diff
@@ -1398,14 +1398,43 @@ func (proc *HandleT) processJobsForDest(jobList []*jobsdb.JobT, parsedEventList
 	proc.pStatsDBW.Print()
 }
 
-func ConvertToTransformerResponse(events []transformer.TransformerEventT) transformer.ResponseT {
+func ConvertToTransformerResponse(events []transformer.TransformerEventT, filterUnsupportedMessageTypes bool) transformer.ResponseT {
 	var responses []transformer.TransformerResponseT
+	var failedEvents []transformer.TransformerResponseT
+
+	// filter unsupported message types
+	var resp transformer.TransformerResponseT
+	var errMessage string
 	for _, event := range events {
-		resp := transformer.TransformerResponseT{Output: event.Message, StatusCode: 200, Metadata: event.Metadata}
-		responses = append(responses, resp)
+		destinationDef := event.Destination.DestinationDefinition
+		supportedTypes, ok := destinationDef.Config["supportedMessageTypes"]
+		if ok && filterUnsupportedMessageTypes {
+			messageType, typOk := event.Message["type"].(string)
+			if !typOk {
+				// add to FailedEvents
+				errMessage = "Invalid message type. Type assertion failed"
+				resp = transformer.TransformerResponseT{Output: event.Message, StatusCode: 400, Metadata: event.Metadata, Error: errMessage}
+				failedEvents = append(failedEvents, resp)
+				continue
+			}
+
+			if misc.Contains(supportedTypes, messageType) {
+				resp = transformer.TransformerResponseT{Output: event.Message, StatusCode: 200, Metadata: event.Metadata}
+				responses = append(responses, resp)
+			} else {
+				// add to FailedEvents
+				errMessage = "Message type " + messageType + " not supported"
+				resp = transformer.TransformerResponseT{Output: event.Message, StatusCode: 400, Metadata: event.Metadata, Error: errMessage}
```

Comment:
Should we set a different field in response struct, so that it is saved so in the proc_err tables and can be also used for use cases like filter out when sending events to live events (if we do so), skip in replay etc

Reply:
save or not to proc_error will be taken up in another PR.